tests/rptest/scale_tests/audit_log_test.py

```diff
@@ -0,0 +1,412 @@
+
+        1. Have the same API type (so kafka::<produce>, or admin_api::<endpoint>)
+        2. Have the same authorization result (so user, ip, credentials, etc)
+        3. Have the same request parameters
+
+        For point (3) the detail of request parameters is usually not so exhaustive, for
+        example in a produce request only the topics not the partitions are part of the
+        audit payload. This means increasing the number of partitions will introduce more
+        traffic only up until a certain point. Only when partitions for the same topic
+        reside on different cores will they produce distinct auditable events.
+
+        Due to the above this test will use many clients with many topics (and a light
+        number of partitions, enough to have 1 per shard) to generate a many
+        unique auditable messages from the produce, consume and authentication
+        APIs.
+        """
+
+        # Scale tests are not run on debug builds
+        assert not self.debug_mode
+
+        # This user will be passed to the kgo-repeater program
+        self._create_audit_test_sasl_user()
+
+        # Message cardinality formula:
+        # distinct_events = number_of_workers * number_of_cores * number_of_topics
+        #
+        # Adding more partitions past the number of cores won't matter because
+        # the partition_id isn't part of the audit message payload. We will want only the
+        # number of partitions up until there is one per core in the entire cluster. To
+        # scale further then that we want to increase the number of topics.
+        scale = ScaleParameters(self.redpanda,
+                                replication_factor=3,
+                                mib_per_partition=DEFAULT_MIB_PER_PARTITION,
+                                partitions_per_shard=PARTITIONS_PER_SHARD)
+        partitions_per_topic = self.redpanda.get_node_cpu_count() * len(
+            self.redpanda.nodes)
+        total_topics = scale.partition_limit // partitions_per_topic
+        extra_partitions = scale.partition_limit % partitions_per_topic
+        if total_topics == 0:
+            total_topics = 1
+            extra_partitions = 0
+
+        # Create the topics as configured with the scale definition above stopping when
+        # the partition limit has been exhausted. The partition limit must be spread across
+        # a number of topics so that no topic has more then partitions_per_topic partitions.
+        topics = [
+            TopicSpec(partition_count=partitions_per_topic)
+            for _ in range(0, total_topics)
+        ]
+        if extra_partitions > 0:
+            # If partition_limit doesn't evenly divide into the total number of shards,
+            # create an extra topic with the number of partitions containing the diff
+            difference = partitions_per_topic - extra_partitions
+            topics = topics + [TopicSpec(partition_count=difference)]
```

Comment:
question: Shouldn't partition count be `extra_partitions`?

Say you have a partition limit of 103 and you have 10 partitions per topic, you'd have 3 partitions left over from `scale.partition_limit % partitions_per_topic`

Reply:
yeah but i am trying to add `7` to make `110` in this case, but now that i think about it, maybe I shouldn't because the bandwidth estimates are based on `scale.partition_limit` and i should be making exactly that